test/lib/redis-store-spec.js

```diff
@@ -152,6 +168,32 @@ describe('set', function () {
 });
 
 describe('get', function () {
+  it('should return a promise', function (done) {
+    assert.ok(redisCache.get('foo') instanceof Promise);
+    done();
+  });
+
+  it('should resolve promise on success', function (done) {
+    redisCache.set('foo', 'bar')
+      .then(() => redisCache.get('foo'))
+      .then(result => {
+        assert.equal(result, 'bar');
+        done();
+      });
+  });
+
+  it('should reject promise on error', function (done) {
+    var pool = redisCache.store._pool;
+    sinon.stub(pool, 'acquireDb').yieldsAsync('Something unexpected');
+    sinon.stub(pool, 'release');
+    redisCache.get('foo')
+      .catch(() => done())
+      .then(() => {
+        pool.acquireDb.restore();
+        pool.release.restore();
```

Comment:
No, tests shouldn't fail if they reach that `then`. A `then` after a `catch` will execute on both resolved and rejected Promises.

The code you posted does the same thing but is much more verbose, I can change it if you want.